server.js

```diff
@@ -1,30 +1,99 @@
 import express from "express";
 import cors from "cors";
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
+import booksData from "./data/books.json";
+
 const port = process.env.PORT || 8080;
 const app = express();
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.json(
+    {
+      "Open API": "Books",
+      "Routes": [
+        {
+          "/books": "Get all books",
+          "/books/:id": "Get book whose id matches",
+          "/ratings/:rating": "Get books whose rating starts with the number provided"
+        }
+      ],
+      "Queries": [
+        {
+          "/books?author=string": "Get all books whose author contains the string provided",
+          "/books?title=string": "Get all books whose title contains the string provided",
+        }
+      ]
+    }
+  );
+});
+
+app.get('/books', (req, res) => {
+	const { author, title } = req.query;
+	let filteredBooks = booksData;
+	
+	if(author) {
+    console.log(filteredBooks.length);
+		filteredBooks = filteredBooks.filter((item) => 
+      item.authors.toLowerCase()
+      .includes(author.toLowerCase()));
+	};
+
+	if(title) {
+		filteredBooks = filteredBooks.filter((item) => 
+			item.title.toLowerCase()
+			.includes(title.toLowerCase()))
+	};
+	
+	if (filteredBooks.length === 0) {
+		res.status(404).json({
+      data: "Sorry we couldn't find book you seek",
+      success: false
+    });
+	} else {
+		res.status(200).json({
+      data: filteredBooks,
+      success: true
+    })
+	};
+});
+
+app.get("/books/:id/", (req, res) => {
+  const bookByID = booksData.find(
+    (book) => book.bookID === parseInt(req.params.id));
+
```

Comment:
Looks great! I used .toString() for my year of realese const so will read up on the parseInt to learn the difference!